### PR TITLE
Deployment scripts rework

### PR DESCRIPTION
**Pull request by bakhtin**
### Description of the Change
- Deployment scripts (Ansible playbook) are reworked without platform-dependent binaries in it
- Squashed different deployment scenarios into a single role
- Remove older Ansible roles
- Update role's README file

Key generation process is now implemented with a Python script. There is also a Bash script (`config_gen.sh`) to generate Iroha keys and `genesis.block` that is executed from inside an Ansible task.

**UPDATE:** Fixed deployment scripts to support RC2

### Benefits
Platform-independent Ansible role that considers multiple deployment scenarios (including local host)

### Possible Drawbacks 
None

### Usage Examples or Tests *[optional]*
See README.md

### CODE REVIEW

**Comment by neewy** [comment]
Tried to run it locally on ubuntu server 18.04.1, failed to run. I am checking what is happening at the moment with my environment. It would be nice to have requirements.txt file for python dependencies

File: deploy/ansible/roles/iroha-docker/README.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+The first one is easier to implement since it does not require preliminary configuration of the remote hosts. Just make sure that network ports are not firewalled. You can check the port list in the generated Docker Compose file (`docker-compose.yml`) after deployment.
+
+The second one can be used when there is an overlay network exists between the hosts. In short, overlay network allows for Docker containers to communicate using a single subnet. Such that each container would have a unique IP address in that subnet. Learn more in official Docker documentation (https://docs.docker.com/network/overlay). We recommend to use Calico for setting up Docker overlay network since it can be used as a network plugin (https://docs.projectcalico.org/v1.5/getting-started/docker/tutorials/basic).
+
+The second way is also suitable for local-only deployments and enabled by default.
+
+## Requirements
+  Tested on Ubuntu 16.04, 18.04
+  - Local:
+    - python-3, python3-dev
```
Comment (neewy) [inline deploy/ansible/roles/iroha-docker/README.md]:
```suggestion
    - python3, python3-dev
```

**Comment by l4l** [comment]
dco :(

**Comment by bakhtin** [comment]
Updated playbook to deploy on the remote host by default (previously, it was set up to deploy locally). Corrected README file to reflect these changes. @neewy you may want to follow role's README to run the role once more.

**Comment by sudomann** [comment]
> There is also a Bash script (config_gen.sh) to generate Iroha keys and genesis.block that is executed from inside an Ansible task.
🥇 
Man, you did my work for me already. I appreciate your efforts!

File: deploy/ansible/roles/iroha-docker/defaults/main.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+# Affects how Iroha peers are communicated. If set to `true`, Docker overlay network with that
+# name will be used. It must be created beforehand. The recommended way is to use Calico plugin
+# for Docker (projectcalico.org).
+# Setting it to `true` without creating the network will only work for a single host deployment
+# (Iroha peers will only be able to communicate within that host). Suitable for local-only
+# deployments for testing purposes.
+#
+# Default: false
+overlay_network: true
```
Comment (neewy) [inline deploy/ansible/roles/iroha-docker/defaults/main.yml]: false

File: deploy/ansible/roles/iroha-docker/README.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+The second way is also suitable for local-only deployments and enabled by default.
+
+## Requirements
+  Tested on Ubuntu 16.04, 18.04
+  - Local:
+    - python3, python3-dev
+    - PIP modules: future, ansible(>=2.4), sha3(for Python<3.6)
+  - Remote:
+    - Docker (>=17.12)
```
[the remark this answers is not in the document]
Comment (neewy) [reply deploy/ansible/roles/iroha-docker/README.md]: `sudo apt install docker.io`

File: deploy/ansible/roles/iroha-docker/README.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+    **iroha.list**
+    ```
+    [all]
+    192.168.122.109
+    ```
+
+    Put this file into `../../inventory/` directory.
+
+`cd ../../ && ansible-playbook -b -e 'ansible_ssh_user=ubuntu' -i inventory/iroha.list playbooks/iroha-docker/main.yml`
```
[the remark this answers is not in the document]
Comment (bakhtin) [reply deploy/ansible/roles/iroha-docker/README.md]: Added an explanation

File: deploy/ansible/roles/iroha-docker/README.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    ```
+    replicas: 2
+    ```
+
+    **192.168.122.30.yml**
+    ```
+    replicas: 4
+    ```
+
+4. Create a playbook:
```
[the remark this answers is not in the document]
Comment (bakhtin) [reply deploy/ansible/roles/iroha-docker/README.md]: not required, removed

**Pull request merged**